src/Scheduler/StatefulScheduledCommandsResolver.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console\Scheduler;
+
+use DateTime;
+use Tempest\Console\ConsoleCommand;
+
+final class StatefulScheduledCommandsResolver implements ScheduledCommandsResolver
+{
+    public const string CACHE_PATH = __DIR__ . '/last-schedule-run.cache.php';
+
+    public function __construct(
+        private SchedulerConfig $config,
+    ) {
+    }
+
+    public function resolve(DateTime $date): array
+    {
+        $lastRuns = $this->get();
```

Comment:
`get` should probably be renamed to `getPreviousRuns()`. I'd also rename `lastRuns` to `previousRuns`